### PR TITLE
Change from BackAndroid to BackHandler

### LINKED ISSUE #1330: BackAndroid is deprecated [react-native 0.44.0]
In react native 0.44.0, BackAndroid is deprecated. Got a warn :/

Thanks !

### PR DESCRIPTION
**Pull request by zackify**
closes #1330 

If someone else could double check this that would be awesome, but looks like the completely same syntax, just a different name!

### CODE REVIEW

**Comment by mica16** [comment]
Got a warning too, the replacement would be cool :) 
Thanks

File: src/createNavigationContainer.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ type NavigationContainerProps = {
   onNavigationStateChange?: (
     NavigationState,
     NavigationState,
-    NavigationAction,
+    NavigationAction
```
Comment (dominicrj23) [inline src/createNavigationContainer.js]: Shouldn't we avoid making unnecessary changes in the code structure? 
Applies to a few other places too.

**Comment by nihgwu** [comment]
so you are going to drop the support for RN < 0.44?

**Comment by zackify** [comment]
Sorry that was due to prettier. I will fix it

**Comment by jakecraige** [comment]
Any update from the core team on this one? This one is frustrating to have with RN 0.44 since it throws a `console.warn` which comes up every time you load the app in dev.

**Comment by weby** [comment]
As a temporary workaround, you can add this line to your App.js to prevent yellow box messages.

`console.ignoredYellowBox = ['Warning: BackAndroid']`

**Comment by ugiacoman** [comment]
@weby Thanks!

File: src/createNavigationContainer.js (hunk 1 above)
Comment (cooperka) [inline src/createNavigationContainer.js]: @zackify These trailing commas are intentional; they help give better diffs when adding/removing/reorganizing things in a list of items. Please keep them in :)

**Comment by zackify** [comment]
I know I totally agree. In fact I have trailing commas turned on in my atom
plugin, don't know how it happened. I'll try to fix the commit tomorrow.

**Comment by m1ckm** [comment]
Ok if it's a need and I can fix tell me what to do but I wasn't impressed with all the up shit shoot and gay stuff. That was being poorly translated.

File: src/createNavigationContainer.js (hunk 1 above)
Comment (zackify) [inline src/createNavigationContainer.js]: Yes... somehow prettier had trailing commas off on my work computer!! My bad.

**Comment by zackify** [comment]
Okay, I fixed the commas!

**Comment by satya164** [comment]
Travis is failing because `BackHandler` is undefined. You also need to update the RN version probably?

**Comment by dominicrj23** [comment]
I don't think we should merge this until we bump react-native

**Comment by SNNikitin** [comment]
How about news?

**Comment by mettin** [comment]
@dominicrj23 this is currently blocking me from updating my project to .44. 
I think its save to assume that people who are upping their react-navigation won't by using an older RN version?

**Comment by satya164** [comment]
> this is currently blocking me

It's a deprecation warning. It shouldn't block you. You can just suppress the warning.

**Comment by skevy** [comment]
Sorry Zach! I didn't see this one and already reviewed #1520.

Thanks for your work here. Hope to see more contributions from you! :)
